alacritty_terminal/src/renderer/rects.rs

```diff
@@ -110,33 +107,52 @@ impl Rects {
                     && cell.column == line.end.col + 1
                 {
                     // Update the length of the line
-                    line.update_end(cell.into(), size);
-
+                    line.end = cell.into();
                     continue;
                 }
             }
 
             // Start new line if there currently is none
-            let rect = Line::from_cell(cell, *flag, metrics, size);
+            let line = Line { start: cell.into(), end: cell.into(), color: cell.fg };
             match self.inner.get_mut(flag) {
-                Some(lines) => lines.push(rect),
+                Some(lines) => lines.push(line),
                 None => {
-                    self.inner.insert(*flag, vec![rect]);
+                    self.inner.insert(*flag, vec![line]);
                 },
             }
         }
     }
+}
+
+/// Generall purpose rects.
+#[derive(Default)]
+pub struct Rects {
+    inner: Vec<(Rect<f32>, Rgb)>,
+}
+
+impl Rects {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Convert the stored rects to rectangles for the renderer.
+    pub fn rects(&self) -> &Vec<(Rect<f32>, Rgb)> {
```

Comment:
Instead of using a getter, I'd usually recommend just making the field `pub`. However this shouldn't be relevant anymore if we decide to add color to `Rect`.

Reply:
I don't think I agree with this. Defining a non `mut` getter with a private field allows the library developer to be sure that that field is never mutated from the outside, whereas a `pub` field can be mutated if it's binding is `mut`. A somewhat small thing, but I find it useful.

Reply:
Is there any good reason why we would want to prevent that here though? That's precisely why we do have `mut` to mark things as mutable or immutable.

Currently this just adds five lines of code which would be completely identical to just accessing it directly.

Reply:
FYI, this code will be deleted in a moment.